### PR TITLE
LoggingClient mods for remote vs file logging

### PR DESCRIPTION
**Pull request by tsconn23**
Changes to the logging client library for:

-    Addition of flag in constructor indicating remote logging enabled.

-   Interpretation of general LogTarget property based on above. If true, then target will be URL. If false, target will be file. These values are supplied by the caller.

-    If logging to file, check to see if directory exists first. If not, create it (I intentionally avoided holding this as a global variable).

-    Lexical modifications to standardize variable names according to conventions in the "Go Programming Language" book

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -27,65 +27,84 @@ import (
...
-	owningServiceName string
-	RemoteUrl         string
-	LogFilePath       string
-	stdOutLogger      *log.Logger
-	fileLogger        *log.Logger
+	OwningServiceName string
+	RemoteEnabled     bool
+	LogTarget         string
+	StdOutLogger      *log.Logger
+	FileLogger        *log.Logger
```
Comment (feclare) [inline logger.go:39]: if the struct fields are not needed outside of the package they should be private, starting with lowercase.

File: logger.go [hunk 2]
```diff
@@ -27,65 +27,84 @@ import (
...
 }
 
 // Create a new logging client for the owning service
-func NewClient(owningServiceName string, remoteUrl string) LoggingClient {
+func NewClient(owningServiceName string, isRemote bool, logTarget string) LoggingClient {
```
Comment (feclare) [inline logger.go:43]: I think it is more readable to keep two different fields to have the remoteUrl and the local log filename. And just enable local and/or remote just if they are not empty.

File: logger.go [hunk 3]
```diff
@@ -27,65 +27,84 @@ import (
...
-		defer file.Close()
-		if err != nil {
-			fmt.Println("Error opening log file: " + err.Error())
-		}
+func (lc LoggingClient) saveToLogFile(prefix string, message string) error {
+	if lc.LogTarget == "" {
+		return nil
+	}
+
+	verifyLogDirectory(lc.LogTarget)
```
Comment (feclare) [inline logger.go:77]: Can be the call to verifyLogDirectory to NewClient?

**Review by feclare — CHANGES_REQUESTED** [review]

File: logger.go (hunk 2 above)
Comment (tsconn23) [reply logger.go:43]: However, in the configs both fields are both populated. Example from core-data-go:

"Loggingfile" : "./core-data.log",
"Loggingremoteurl" : "http://localhost:48061/api/v1/logs"

So an additional boolean is necessary.

File: logger.go (hunk 1 above)
Comment (tsconn23) [reply logger.go:39]: OK, will change this

File: logger.go (hunk 3 above)
Comment (tsconn23) [reply logger.go:77]: You're saying put this in the constructor? OK, let me look at doing that.

**Pull request closed without merging**